packages/bot-skeleton/src/services/tradeEngine/trade/Purchase.js

```diff
@@ -42,37 +38,53 @@ export default Engine =>
                     buy_price: buy.buy_price,
                 });
             };
-            const action = () => api_base.api.send({ buy: id, price: askPrice });
-            this.isSold = false;
-            contractStatus({
-                id: 'contract.purchase_sent',
-                data: askPrice,
-            });
 
-            if (!this.options.timeMachineEnabled) {
-                return doUntilDone(action).then(onSuccess);
+            function hasBlockOfType(targetType, workspace) {
+                const allBlocks = workspace.getAllBlocks();
+                return allBlocks.some(block => block.type === targetType);
+            }
+
+            const workspace = Blockly.getMainWorkspace();
+            const hasPayoutBlock = hasBlockOfType('payout', workspace);
```

Comment:
![image](https://github.com/binary-com/deriv-app/assets/97010868/ceaf11ae-c38e-45b5-8084-890cc5f3386e)

we may also need a check to check if the payout block is inside any parent block and kept outside parent block.
if kept inside strategy me need to set the flag `true`

Reply:
Yes, only in case the payout block is outside of the parent main block and misplaced I suppose we don't consider the block a part of the running strategy

Reply:
has included check presences of parent block of payout block ✅